recipes/cuda-cudart/meta.yaml

```
@@ -0,0 +1,290 @@
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: CUDA Runtime native Libraries
+      description: |
+        CUDA Runtime native Libraries
+      doc_url: https://docs.nvidia.com/cuda/index.html
+
+  - name: cuda-cudart-dev_{{ target_platform }}
+    build:
+      noarch: generic
+      run_exports:
+        - {{ pin_subpackage("cuda-cudart", max_pin="x") }}
+    files:
+      - targets/{{ target_name }}/include/*                     # [linux]
+      - targets/{{ target_name }}/include/cooperative_groups/*  # [linux]
+      - targets/{{ target_name }}/lib/*.so                      # [linux]
+      - Library\include                                         # [win]
+      - Library\lib\cuda.lib                                    # [win]
+      - Library\lib\cudadevrt.lib                               # [win]
+      - Library\lib\cudart.lib                                  # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - cuda-cccl_{{ target_platform }}
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - cuda-cudart-static_{{ target_platform }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      commands:
+        - test -f $PREFIX/targets/{{ target_name }}/include/cuda.h              # [linux]
+        - test -f $PREFIX/targets/{{ target_name }}/include/cuda_runtime.h      # [linux]
+        - test -f $PREFIX/targets/{{ target_name }}/include/cuda_runtime_api.h  # [linux]
+        - test -L $PREFIX/targets/{{ target_name }}/lib/libcudart.so            # [linux]
+        - if not exist %LIBRARY_INC%\cuda.h exit 1                              # [win]
+        - if not exist %LIBRARY_INC%\cuda_runtime.h exit 1                      # [win]
+        - if not exist %LIBRARY_INC%\cuda_runtime_api.h exit 1                  # [win]
+        - if not exist %LIBRARY_LIB%\cuda.lib exit 1                            # [win]
+        - if not exist %LIBRARY_LIB%\cudadevrt.lib exit 1                       # [win]
+        - if not exist %LIBRARY_LIB%\cudart.lib exit 1                          # [win]
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: CUDA Runtime native Libraries
+      description: |
+        CUDA Runtime native Libraries
+      doc_url: https://docs.nvidia.com/cuda/index.html
+
+  - name: cuda-cudart-static
```

It seems this package is not used on Windows, I think we should skip it:
```suggestion
  - name: cuda-cudart-static
    build:
      skip: True  # [not linux]
```

---

This package is here on Windows for consistency. It would still pull in `cudart_static.lib` through `cuda-cudart-static_{{ target_platform }}`

---

Yes, it is a pure metapackage (no contents, only a dependency). We should keep it.

---

But this is confusing. This package is only used in `cuda-cudart-dev`, but there's an asymmetry: it depends on `cuda-cudart-dev_{{ target_platform }}` (line 83), but not on `cuda-cudart-static_{{ target_platform }}` (line 85) which would have eliminated this asymmetry. Why is it the case? Am I missing something?

---

I think this package needs to be moved back to `run` and enabled for windows in `cuda-cudart-dev`

https://github.com/conda-forge/staged-recipes/pull/21723/files#r1158680184

---

Wait, I think we agreed that `-static` packages are placed in `run_constrained` not `run`, so that if users only need headers, they don't have to pull a big thing? What's changed?

---

`cudart` was the exception because of `nvcc` static linking to it.

---

Ah ok... 😞